pandas/core/indexes/base.py

```diff
@@ -3619,7 +3619,12 @@ def _get_indexer(
         elif method == "nearest":
             indexer = self._get_nearest_indexer(target, limit, tolerance)
         else:
-            indexer = self._engine.get_indexer(target._get_engine_target())
+            tgt_values = target._get_engine_target()
+            if target._is_multi and self._is_multi:
```

Comment:
might be worth a comment here and L5453 that this is for perf as not super obvious